### DISCUSSION TITLE
adding IUserRetrieveService in IRequestContext

### DISCUSSION THREAD

**Discussion by dfaruque** [serenity-is/Serenity#5859] [Serenity Ideas]
There are many users of Serenity who face difficulty in fetching user definitions in the .net 5 versions of serenity. Because, if we need user definition in the repository then we need to add a parameter for IUserRetrieveService and pass that from the endpoint. It will be very convenient if we have IUserRetrieveService in IRequestContext, so that we don't need to add another parameter in the constructors for just retrieving the user definition. The code will be much simpler like the following 
`var userDefinition = User.GetUserDefinition(Context.UserRetrieveService);`

**Comment by chaurasiaakash204** [comment]
Please send the sample code

**Comment by VictorTomaili** [comment]
We have some plans for that. But we will prefer to work on claims instead add IUserRetrieveService. User claims are the more modernized way to access the current user.

**Reply by dfaruque** [reply]
Thanks for the reply. The way could be anything. We only need to access user-definition on the fly like previous versions of serentiy.
